src/main/kotlin/com/sourcegraph/cody/agent/CodyAgentService.kt

```diff
@@ -251,7 +251,8 @@ class CodyAgentService(private val project: Project) : Disposable {
         restartIfNeeded: Boolean,
         callback: Consumer<CodyAgent>,
         onFailure: Consumer<Exception> = Consumer {}
-    ) {
+    ): CompletableFuture<Boolean> {
```

Comment:
Another facility that can be useful for testing. I was using it extensively for a while. I wound up redoing those tests, but I still think it might be useful.